src/VisualStudio/Core/Def/Implementation/DesignerAttribute/DesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -69,21 +70,47 @@ public bool NeedsReanalysisOnOptionChanged(object sender, OptionChangedEventArgs
             return false;
         }
 
-        public async Task AnalyzeDocumentAsync(Document document, SyntaxNode bodyOpt, InvocationReasons reasons, CancellationToken cancellationToken)
+        public async Task AnalyzeProjectAsync(Project project, bool semanticsChanged, InvocationReasons reasons, CancellationToken cancellationToken)
         {
-            Contract.ThrowIfFalse(document.IsFromPrimaryBranch());
-
             cancellationToken.ThrowIfCancellationRequested();
 
-            if (!document.Project.Solution.Workspace.Options.GetOption(InternalFeatureOnOffOptions.DesignerAttributes))
+            if (!project.Solution.Workspace.Options.GetOption(InternalFeatureOnOffOptions.DesignerAttributes))
             {
                 return;
             }
 
+            var projectVersion = await project.GetDependentVersionAsync(cancellationToken).ConfigureAwait(false);
+
+            try
+            {
+                // Process all the documents in this project in parallel.  The project system can
+                // handle multiple calls into SetProjectItemDesignerTypeAsync and will batch them
+                // appropriately.
```

Comment:
note: this is how things were basically working before as well.  We were fire-and-forget in the other PR.  Now, i'm just being more explicit about that, and providing a single place to handle all the tasks we fire off.

Reply:
project analysis runs at the end of all work. if we can do things as user type, that is better in my opinion. no reason to make things slower just to make code look good.

Reply:
by the way, I update PR last night to do await but in document analysis.